### ISSUE TITLE
Account verifying seems flawed

### ISSUE THREAD

**Issue by sparik** [fxfactorial/silicondzor#15]
I think [this](https://github.com/fxfactorial/silicondzor/blob/7ee054461f44f94989fa6410ea11dc5b9f15b74f/backend/server.js#L58) is supposed to be a setInterval instead of setTimeout. Otherwise it will clear the object only once, then the object will keep growing. 
Even with that fix, an account verification link may be valid not for 1 hour, but for _up to_ 1 hour. It may be valid for 1 second if a user registers 1 second before the register_email_users cleanup. This may be fixed by keeping a queue of registered users, pushing when a user registers and the link is sent, and poping all expired links every 1 hour, for example. Also this will make the implementation of 're-send the link' easier.

**Comment by fxfactorial** [comment]
Oops! yea, was supposed to be setInterval. 

Yea I thought about the up to 1 hour issue, keeping a queue is a fancier version of what I imagined as the fix which would have been just to iterate the whole thing every hour. 

Can you make a PR fixing the `setInterval` mistake?

**Comment by sparik** [comment]
I'm new to this. I'll try now

**Comment by sparik** [comment]
@fxfactorial done

**Comment by aantron** [comment]
Was fixing all of this; but shouldn't we store this in the database. so the links set will survive a server restart?

**Comment by sparik** [comment]
Yes, I think the link along with its creation date (for checking if it's expired) should be stored in the database.

**Comment by fxfactorial** [comment]
Yes, that's the correct thing to do

**Comment by aantron** [comment]
Ok, already have it locally :) will PR later.
